src/main/java/com/mraof/minestuck/client/gui/AreaEffectScreen.java

```diff
@@ -29,63 +34,112 @@ public class AreaEffectScreen extends Screen
-		this.minPosDestinationTextFieldY.setValue(String.valueOf(te.getMinEffectPos().getY()));
+		this.minPosDestinationTextFieldY.setValue(String.valueOf(te.getMinAreaOffset().getY()));
 		addButton(minPosDestinationTextFieldY);
 		
 		this.minPosDestinationTextFieldZ = new TextFieldWidget(this.font, this.width / 2 + 20, yOffset + 15, 40, 20, new StringTextComponent("Z value of min effect pos"));
-		this.minPosDestinationTextFieldZ.setValue(String.valueOf(te.getMinEffectPos().getZ()));
+		this.minPosDestinationTextFieldZ.setValue(String.valueOf(te.getMinAreaOffset().getZ()));
 		addButton(minPosDestinationTextFieldZ);
 		
 		this.maxPosDestinationTextFieldX = new TextFieldWidget(this.font, this.width / 2 - 60, yOffset + 50, 40, 20, new StringTextComponent("X value of max effect pos"));
-		this.maxPosDestinationTextFieldX.setValue(String.valueOf(te.getMaxEffectPos().getX()));
+		this.maxPosDestinationTextFieldX.setValue(String.valueOf(te.getMaxAreaOffset().getX()));
 		addButton(maxPosDestinationTextFieldX);
 		
 		this.maxPosDestinationTextFieldY = new TextFieldWidget(this.font, this.width / 2 - 20, yOffset + 50, 40, 20, new StringTextComponent("Y value of max effect pos"));
-		this.maxPosDestinationTextFieldY.setValue(String.valueOf(te.getMaxEffectPos().getY()));
+		this.maxPosDestinationTextFieldY.setValue(String.valueOf(te.getMaxAreaOffset().getY()));
 		addButton(maxPosDestinationTextFieldY);
 		
 		this.maxPosDestinationTextFieldZ = new TextFieldWidget(this.font, this.width / 2 + 20, yOffset + 50, 40, 20, new StringTextComponent("Z value of max effect pos"));
-		this.maxPosDestinationTextFieldZ.setValue(String.valueOf(te.getMaxEffectPos().getZ()));
+		this.maxPosDestinationTextFieldZ.setValue(String.valueOf(te.getMaxAreaOffset().getZ()));
 		addButton(maxPosDestinationTextFieldZ);
 		
-		addButton(new ExtendedButton(this.width / 2 - 20, yOffset + 73, 40, 20, new StringTextComponent("DONE"), button -> finish()));
+		
+		this.effectTextField = new TextFieldWidget(this.font, this.width / 2 - 65, yOffset + 79, 105, 18, new StringTextComponent("Current Effect"));
+		this.effectTextField.setValue(te.getEffect().getRegistryName().toString());
+		addButton(effectTextField);
+		
+		this.effectAmplifierTextField = new TextFieldWidget(this.font, this.width / 2 + 45, yOffset + 79, 20, 18, new StringTextComponent("Current Effect Amplifier"));
+		this.effectAmplifierTextField.setValue(String.valueOf(te.getEffectAmplifier()));
+		addButton(effectAmplifierTextField);
+		
+		if(isAllMobs)
+			addButton(allMobsButton = new ExtendedButton(this.width / 2 - 65, yOffset + 105, 85, 20, new StringTextComponent("ALL MOBS"), button -> cycleIsAllMobs()));
+		else
+			addButton(allMobsButton = new ExtendedButton(this.width / 2 - 65, yOffset + 105, 85, 20, new StringTextComponent("JUST PLAYERS"), button -> cycleIsAllMobs()));
```

Comment:
`isAllMobs` only determines the text on the button. With this in mind, you can simplify this part of the code.

Reply:
With a helper function
```java
private void updateAllMobsButtonMessage()
{
	if(isAllMobs)
		allMobsButton.setMessage(new StringTextComponent("ALL MOBS"));
	else
		allMobsButton.setMessage(new StringTextComponent("JUST PLAYERS"));
}
```
you can simplify the initialization of the button to
 ```java
 addButton(allMobsButton = new ExtendedButton(this.width / 2 - 65, yOffset + 105, 85, 20, StringTextComponent.EMPTY, button -> cycleIsAllMobs()));
 updateAllMobsButtonMessage();
 ```
 And also use that helper function in `cycleIsAllMobs()` to make sure that the button message is only defined in one place.